spec/bolt_server/transport_app_spec.rb

```diff
@@ -411,91 +333,67 @@ def mock_plan_info(full_name)
         }
 
         it '/tasks/:module_name/:task_name handles task name = module name (init.rb) task' do
-          expect(BoltServer::PE::PAL).to receive(:new).and_return(fake_pal)
-          expect(fake_pal).to receive(:get_task).with(task_name).and_return(mock_task)
           get(path)
           resp = JSON.parse(last_response.body)
           expect(resp).to eq(expected_response)
         end
       end
-
-      context 'with non-existant task' do
-        let(:path) { '/tasks/foo/bar?environment=production' }
-        it 'returns 400 if an unknown plan error is thrown' do
-          expect(BoltServer::PE::PAL).to receive(:new).and_return(fake_pal)
-          expect(fake_pal).to receive(:get_task).with('foo::bar').and_raise(Bolt::Error.unknown_task('foo::bar'))
-          get(path)
-          expect(last_response.status).to eq(400)
-        end
-      end
```

Comment:
In general in this file I think its good that we are not mocking PAL. Maybe i'm missing it but it looks like we lost all our tests around non-existant content?

Reply:
I removed several of these because they are redundant. Most of the functionality uses the same helper functions, and it didn't seem particularly helpful to retest those helpers over and over.